lrmd/tls_backend.c

```diff
@@ -260,6 +261,24 @@ lrmd_remote_listen(gpointer data)
         return TRUE;
     }
 
+#ifdef TCP_USER_TIMEOUT
+    {
+    int rc;
+    int optval;
+    long sbd_timeout = get_sbd_timeout();
+
+    if (sbd_timeout > 0) {
+        optval = sbd_timeout / 2; /* time to fail and retry before watchdog */
+        rc = setsockopt(csock, SOL_TCP, TCP_USER_TIMEOUT, &optval, sizeof(optval));
+        if (rc < 0) {
+            crm_perror(LOG_INFO, "writing TCP_USER_TIMEOUT (%d) on client socket failed", optval);
+            close(csock);
+            return TRUE;
+        }
+    }
+    }
+#endif
+
```

Comment:
Let's functionize this block in lib/common, then we can make get_sbd_timeout() static inline.

Any exposed symbols, even if only used internally, should start with crm_. (Exempting existing symbols of course, for backward API compatibility.)

The functions (including check_sbd_timeout()) should really be in lib/common/watchdog.c. I know the other check_*() functions are in utils.c, but it seems the watchdog aspect is more relevant here.